api/applications.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-multierror"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	msgraphsdkgo "github.com/microsoftgraph/msgraph-sdk-go"
+	auth "github.com/microsoftgraph/msgraph-sdk-go-core/authentication"
+	"github.com/microsoftgraph/msgraph-sdk-go/applications"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+)
+
+type ApplicationsClient interface {
+	GetApplication(ctx context.Context, clientID string) (models.Applicationable, error)
+	CreateApplication(ctx context.Context, displayName string) (models.Applicationable, error)
+	DeleteApplication(ctx context.Context, applicationObjectID string, permanentlyDelete bool) error
+	ListApplications(ctx context.Context, filter string) ([]models.Applicationable, error)
+	AddApplicationPassword(ctx context.Context, applicationObjectID string, displayName string, endDateTime time.Time) (models.PasswordCredentialable, error)
+	RemoveApplicationPassword(ctx context.Context, applicationObjectID string, keyID *uuid.UUID) error
+}
+
+var _ ApplicationsClient = (*AppClient)(nil)
+var _ GroupsClient = (*AppClient)(nil)
+var _ ServicePrincipalClient = (*AppClient)(nil)
+
+type AppClient struct {
```

Comment:
I wonder if `MSGraphClient` is a better name for this? It looks like it's the receiver for methods throughout apps, groups, SPs.